astropy/visualization/scripts/fits2bitmap.py

```diff
@@ -71,12 +71,30 @@ def fits2bitmap(filename, ext=0, out_fn=None, scale='linear',
     import matplotlib.cm as cm
     import matplotlib.image as mimg
 
-    hdulist = fits.open(filename)
-    image = hdulist[ext].data
-    hdulist.close()
+
+    # __main__ gives ext as a string
+    try:
+        ext = int(ext)
+    except ValueError:
+        pass
+
+    try:
+        image = getdata(filename, ext)
+    except Exception as e:
+        log.critical(e)
+        return
+
+    if image.ndim != 2:
+        log.critical('data in FITS extension {0} is not a 2D array'
+                     .format(ext))
+
     if out_fn is None:
-        out_fn = filename.replace('.fits', '.png')
-    if cmap not in cm.datad.keys():
+        out_fn = os.path.splitext(filename)[0]
+        if out_fn.endswith('.fits'):
+            out_fn = os.path.splitext(out_fn)[0]
+        out_fn += '.png'
+
+    if cmap not in cm.datad:
         log.critical('{0} is not a valid matplotlib colormap '
                      'name'.format(cmap))
         raise SystemExit()
```

Comment:
I realize this was already here, but instead of `raise SystemExit()` this should also just be a `return`.

Though in both cases where an error occurred I would change the bare `return` to `return 1` to indicate an error (the return value of the main function is passed to `sys.exit()` by the script wrapper).